framework/JobHandler.py

```diff
@@ -312,12 +313,17 @@ def addJob(self, args, functionToRun, identifier, metadata=None, forceUseThreads
         this runner. For example, if present, to retrieve this runner using the
         method jobHandler.getFinished, the uniqueHandler needs to be provided.
         If uniqueHandler == 'any', every "client" can get this runner
+      @ In, groupInfo, dict, optional, {id:string, size:int}.
+        - "id": it is a special keyword attached to
+          this runner to identify that this runner belongs to a special set of runs that need to be
+          grouped together (all will be retrievable only when all the runs ended).
+        - "size", number of runs in this group
       @ In, clientQueue, boolean, optional, if this run needs to be added in the
         clientQueue
       @ Out, None
     """
     assert "original_function" in dir(functionToRun), "to parallelize a function, it must be" \
```

Comment:
Note that by default, raven_framework runs with debugging turned off so this may not be checked.